### ISSUE TITLE
PnP Provisioning: Content Type Hidden fields does not show in list settings

### ISSUE THREAD

**Issue by umaknow-francis** [SharePoint/PnP-Sites-Core#1899]
#### Category
[X ] Bug

#### Environment
[X ] Office 365 / SharePoint Online

#### Expected or Desired Behavior
Content Type fields marked as "hidden" should still display in the list settings where they are applied.

#### Observed Behavior
When I provision a content type with a hidden field and apply it to a list via a PNP Provisioning template, the field is not visible in the list settings (and the list content type) and cannot be edited in Quick Edit or in a workflow. If I try to add it manually in the browser, the column is not shown, suggesting that it is still there. The field is indeed visible in the edit view page. The column therefore seems to be in a limbo state.

Also, if I get a template from a working site and apply it directly to another, the bug manifests itself.

#### Steps to Reproduce

1. Apply the following template to a vanilla communication site. [list3.zip](https://github.com/SharePoint/PnP-Sites-Core/files/2358184/list3.zip). It was extracted from a working site.
2. Notice the bnc_dc_status ("CI Status") column assigned to the content type as "hidden"
3. The content type is assigned the the OOB Document library
4. Verify that you can see the column in the site content type definition
5. Now see that the column in not visible in the Document list setting and not visible in the *list* content type. It is however visible in the list view.
6. There is also no way to edit the column value (quick edit, SPD workflow)
7. There is no way to add the site column manually to the list content type.

Thank you! I would welcome any workaround.

**Comment by umaknow-francis** [comment]
I just tested to hide the field after the fact using cmdlets:
`$f = Get-PnPField -List "Documents" -Identity "bnc_dc_status"
$f.Hidden = $true
$f.UpdateAndPushChanges($true)`

And I realize that the same bogus behavior happens with that too. It removes completely from the UI the column in list settings and list content type page.

**Comment by IS2** [comment]
I have the same problem publishing content types via ContentTypeHub :-(

**Comment by mag210** [comment]
Is there any update on this? Using the November 2019 release of PNP results in the same issue. What I am trying to achieve is to hide the column from the page details view but still ensure it is present and attached to the list.

**Comment by mmohamedbasith** [comment]
This is not an issue with PNP . This is the default behavior of SharePoint from 2010 . You can hide column in content type , then create a document lib/list and inherit that content type. You will see the hidden column in UI missing in document library. Though it is not desirable , that is how it is.

**Comment by PzKfWg** [comment]
Yes, reading my problem description, I realize that I mixed two different concepts: hiding a field by itself (which hides it from the UI) vs. hiding a field through its relationship with its content type which makes it still appear in the UI (apart from the edit form) which is the desirable behavior. 

The PowerShell I give in my first answer does the first scenario and is expected. It hides the column in its own definition, so it doesn't appear anywhere in the list UI.

In my original post, I hide the field through its relationship with the content type, not the definition.
Definition:        
```
<Field Type="Text" DisplayName="CI Status" Required="FALSE" EnforceUniqueValues="FALSE" Indexed="FALSE" MaxLength="255" Group="BNC - Corporate Instruction" ID="{196b2583-3796-4373-9fc3-3a2c3c7ad0c3}" SourceID="{961aab34-15ee-444a-9be1-946f1e29a156}" StaticName="bnc_dc_status" Name="bnc_dc_status" Description="{res:Field_CI_Status_Description}" CustomFormatter="" AllowDeletion="TRUE">
        </Field>
```
No hiding here. Now the relationship with the CT:
       ```
 <pnp:ContentType ID="0x01010024FD154B97C200469FB7F191AAE872C8" Name="Corporate Instruction" Description="Create a new document." Group="BNC - Corporate Instruction" NewFormUrl="" EditFormUrl="" DisplayFormUrl="">
          <pnp:FieldRefs>
[...]
            <pnp:FieldRef ID="b2f211f1-8289-4eb6-86ec-bdb7f46dd4f8" Name="bnc_dc_reference" />
            <pnp:FieldRef ID="196b2583-3796-4373-9fc3-3a2c3c7ad0c3" Name="bnc_dc_status" **Hidden="true"** />
          </pnp:FieldRefs>
        </pnp:ContentType>
```
This should be the standard UI behavior of hiding a field through its CT relationship, but PNP does it the same as in my PowerShell example, hiding the column by its own definition itself. (I did not retest it since, though. can't say if it is fixed.)
